### PR TITLE
DEV VX-142: Node metrics handler

### PR DESCRIPTION
**Pull request by WakeupTsai**

### CODE REVIEW

File: src/entity/node_metrics.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package entity
+
+type NICNetworkTrafficMetrics struct {
```
Comment (hwchiu) [inline src/entity/node_metrics.go]: add the json structure tag

File: src/server/handler_prometheus.go [hunk 2]
```diff
@@ -17,15 +19,12 @@ func queryMetrics(ctx *web.Context) {
 
 	query := query.New(req.Request.URL.Query())
 
-	query_str := ""
+	expression := ""
```
Comment (John-Lin) [inline src/server/handler_prometheus.go]: how about using `var expression string`?

File: src/server/handler_node_metrics.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	"github.com/linkernetworks/vortex/src/entity"
+	response "github.com/linkernetworks/vortex/src/net/http"
+	"github.com/linkernetworks/vortex/src/web"
+	"github.com/prometheus/common/model"
+)
+
+func listNodeMetricsHandler(ctx *web.Context) {
+	sp, req, resp := ctx.ServiceProvider, ctx.Request, ctx.Response
+
+	result, err := queryFromPrometheus(sp, "sum by (node)(kube_node_info)")
```
Comment (John-Lin) [inline src/server/handler_node_metrics.go]: should handle error. rignt?

File: src/server/handler_node_metrics.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+func getNodeMetricsHandler(ctx *web.Context) {
+	sp, req, resp := ctx.ServiceProvider, ctx.Request, ctx.Response
+
+	id := req.PathParameter("id")
+	node := entity.NodeMetrics{}
+	node.Info.Labels = map[string]string{}
+	node.NICs = map[string]entity.NICMetrics{}
+
+	results, _ := queryFromPrometheus(sp, `{__name__=~"kube_node_info|kube_node_created|node_network_interface|kube_node_labels|kube_node_status_allocatable_cpu_cores|kube_node_status_allocatable_memory_bytes|kube_node_status_capacity_cpu_cores|kube_node_status_capacity_memory_bytes|",node=~"`+id+`"}`)
```
Comment (John-Lin) [inline src/server/handler_node_metrics.go]: should handle error. rignt?

File: src/server/handler_node_metrics.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+		case "kube_node_status_capacity_cpu_cores":
+			node.Resource.CapacityCPU = float32(result.Value)
+
+		case "kube_node_status_capacity_memory_bytes":
+			node.Resource.CapacityMemory = float32(result.Value)
+		}
+	}
+
+	results, _ = queryFromPrometheus(sp, `sum by(__name__, resource) ({__name__=~"kube_pod_container_resource_limits|kube_pod_container_resource_requests",node=~"`+id+`"})`)
```
Comment (John-Lin) [inline src/server/handler_node_metrics.go]: should handle error. rignt?

File: src/server/handler_node_metrics.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+			switch result.Metric["resource"] {
+			case "cpu":
+				node.Resource.CPULimits = float32(result.Value)
+			case "memory":
+				node.Resource.MemoryLimits = float32(result.Value)
+			}
+		}
+	}
+
+	results, _ = queryFromPrometheus(sp, `{__name__=~"node_network_interface|node_network_receive_bytes_total|node_network_transmit_bytes_total|node_network_receive_packets_total|node_network_transmit_packets_total",node=~"`+id+`"}`)
```
Comment (John-Lin) [inline src/server/handler_node_metrics.go]: should handle error. rignt?

File: src/server/handler_prometheus.go [hunk 7]
```diff
@@ -17,15 +19,12 @@ func queryMetrics(ctx *web.Context) {
...
 	if q, ok := query.Str("query"); ok {
-		query_str = q
+		expression = q
 	}
 
-	api := sp.Prometheus.API
-
-	testTime := time.Now()
-	result, err := api.Query(context.Background(), query_str, testTime)
+	result, err := queryFromPrometheus(sp, expression)
```
Comment (John-Lin) [inline src/server/handler_prometheus.go]: should handle error. rignt?

File: src/server/handler_prometheus.go [hunk 8]
```diff
@@ -36,3 +35,18 @@ func queryMetrics(ctx *web.Context) {
 		"results": result,
 	}, restful.MIME_JSON)
 }
+
+func queryFromPrometheus(sp *serviceprovider.Container, expression string) (model.Vector, error) {
+
+	api := sp.Prometheus.API
+
+	testTime := time.Now()
+	result, err := api.Query(context.Background(), expression, testTime)
```
Comment (John-Lin) [inline src/server/handler_prometheus.go]: should handle error. rignt?

**Pull request merged**